src/main/java/com/bookshop/validator/annotation/Isbn.java

```diff
@@ -1,25 +1,22 @@
-package com.bookshop.annotation;
-
-import jakarta.validation.Constraint;
-import jakarta.validation.Payload;
-import jakarta.validation.ReportAsSingleViolation;
-import jakarta.validation.constraints.NotBlank;
-import jakarta.validation.constraints.Pattern;
-import java.lang.annotation.ElementType;
-import java.lang.annotation.Retention;
-import java.lang.annotation.RetentionPolicy;
-import java.lang.annotation.Target;
-import org.hibernate.validator.constraints.ConstraintComposition;
-
-@Target(ElementType.FIELD)
-@Retention(RetentionPolicy.RUNTIME)
-@Constraint(validatedBy = {})
-@ConstraintComposition
-@ReportAsSingleViolation
-@NotBlank
-@Pattern(regexp = "^(?=(?:[^0-9]*[0-9]){10}(?:(?:[^0-9]*[0-9]){3})?$)[\\d-]+$")
-public @interface Isbn {
-    String message() default "";
-    Class<?>[] groups() default { };
-    Class<? extends Payload>[] payload() default { };
-}
+package com.bookshop.validator.annotation;
+
+import com.bookshop.validator.IsbnValidator;
+import jakarta.validation.Constraint;
+import jakarta.validation.Payload;
+import jakarta.validation.ReportAsSingleViolation;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.hibernate.validator.constraints.ConstraintComposition;
+
+@Target(ElementType.FIELD)
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = {IsbnValidator.class})
+@ConstraintComposition
+@ReportAsSingleViolation
```

Comment:
you don't have any additional constraints - therefore you don't need these annotations